src/libsystemd/sd-journal/mmap-cache.c

```diff
@@ -77,11 +77,14 @@ struct MMapCache {
 MMapCache* mmap_cache_new(void) {
         MMapCache *m;
 
-        m = new0(MMapCache, 1);
+        m = new(MMapCache, 1);
         if (!m)
                 return NULL;
 
-        m->n_ref = 1;
+        *m = (MMapCache) {
+                .n_ref = 1,
+        };
+
```

Comment:
This is a pointless increase of line count.

Using designated struct initializers has its place, but this is just not one of them - not when you have a heap-allocated zero-initialized struct on-hand.  I'd argue this commit makes the code objectively worse than it was previously.

What exactly is the advantage of the new form?  What am I missing?  It seems like cargo-cult behavior from where I'm sitting.

Reply:
> It seems like cargo-cult behavior from where I'm sitting.

Maybe. But we usually do like that. I just followed our coding style (https://systemd.io/CODING_STYLE/).
I think it is OK to drop the commit if many of us dislike the change. But,

> I'd argue this commit makes the code objectively worse than it was previously.

this sounds overclaim for me.

Reply:
> > It seems like cargo-cult behavior from where I'm sitting.
> 
> Maybe. But we usually do like that. I just followed our coding style (https://systemd.io/CODING_STYLE/). I think it is OK to drop the commit if many of us dislike the change. But,
> 

The example shown in CODING_STYLE isn't heap-allocated though, and actually makes sense since you eliminate the necessity for zero().

Here the struct must be allocated either way, so just `new0()` makes perfect sense to get the zero-initialized contents.  Then just initialize the members, which in this particular case is just a single member.  The existing code is more succinct.

Maybe the other maintainers disagree with me here, but I'd argue the CODING_STYLE document should be more clear about this.  We don't benefit from this style of initialization on new0-allocated structs IMHO.

Reply:
Let's not bother with specifics too much, if this matters, let's address it in a separate PR for the coding style